controllers/binder.go

```diff
@@ -158,14 +158,26 @@ func (b *binder) updateSpecVolumes(obj *unstructured.Unstructured) error {
 func (b *binder) removeSpecVolumes(
 	obj *unstructured.Unstructured,
 ) (*unstructured.Unstructured, error) {
+	var volumesUnstructured []interface{}
 	volumes, err := b.extractSpecVolumes(obj)
 	if err != nil {
 		return nil, err
 	}
+
 	volumes = b.removeVolumes(volumes)
-	if err = unstructured.SetNestedField(obj.Object, volumes, b.getVolumesPath()...); err != nil {
+
+	for _, v := range volumes {
+		volume, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&v)
+		if err != nil {
+			return nil, err
```

Comment:
how is this change related to the PR scope?

Reply:
One of the unit test was failing -> https://github.com/redhat-developer/service-binding-operator/pull/907/files#diff-e3755e7d9c38983d49f099aa4702dd1afe67677cd5193ea7cdcf03af46c15570R302

The reason for failure was that `unstructured.SetNestedField(...)` was expecting an `interface{}` value. This piece of code converts `corev1.Volume` to `interface{}`

This piece of code is executed when there are additional volumes present in the deployment other than the `ServiceBinding` ones. I have also added a test for this scenario.